### PR TITLE
Fix performance-unnecessary-copy-initialization

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel @mingfeima @XiaobingSuper @ashokei @jingxu10

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/operators/quantization.cpp [hunk 1]
```diff
@@ -598,11 +597,8 @@ Tensor computeQuantizedCat(
     const std::vector<ArgValue>& inputs,
     const std::vector<ExprHandle>& outputShape,
     const std::vector<ExprHandle>& outputStrides,
-    // NOLINTNEXTLINE
     const std::optional<ScalarType>& outputType,
-    // NOLINTNEXTLINE
     at::Device device) {
-  // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
   auto inputList = std::get<BufList>(inputs[0]);
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/operators/quantization.cpp:602]: Not sure if this is actually a fix, or just no longer reported / not reported by the current clang-tidy version. At least I don't see why we need to make an explicit copy here when we only index into the list.

File: torch/csrc/jit/tensorexpr/operators/reduction.cpp [hunk 2]
```diff
@@ -168,7 +168,6 @@ Tensor computeAdaptiveAvgPool2d(
     dtype = Dtype(*outputType);
   }
   BufHandle ResultBuf("adaptive_avgpool2d", outputShape, dtype);
-  // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
   auto out_size_param = std::get<IntList>(inputs[1]);
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/operators/reduction.cpp:171]: Same here as above

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/operators/reduction.cpp (hunk 2 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/operators/reduction.cpp]: Use reference

File: torch/csrc/jit/tensorexpr/operators/quantization.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/operators/quantization.cpp]: Use reference

**Pull request closed without merging**